### ISSUE TITLE
Learning rate scheduler

### ISSUE THREAD

**Issue by joeloskarsson** [mllam/neural-lam#97]
Learning rate scheduling is something useful that we should have as an option. As setting up a way to configure arbitrary pytorch schedulers becomes over-complicated, we should start by just adding in options for cosine schedule. @ErikLarssonDev has a prototype for this.

**Comment by leifdenby** [comment]
In case it is useful: I also made a prototype for this a while back: https://github.com/leifdenby/neural-lam/tree/feat/learning-rate-annealing

**Comment by sadamov** [comment]
@joeloskarsson @sadamov this feature is already part of the research branch.

**Comment by joeloskarsson** [comment]
Specifically, a simple cosine-scheduling is part of the research branch. The approach proposed in #107 is much more extensive. The main question that remains is however how to actually configure the lr schedule for each training run.

**Comment by amanmurari** [comment]
Hi, I would like to work on this issue as part of preparing for GSoC.
Could I be assigned to this?

**Comment by BlackRuffians** [comment]
.take

**Comment by SKM2227229725** [comment]
Hi @joeloskarsson,

I’d like to work on this issue and implement the cosine learning rate scheduler option. I’ll start exploring the current training configuration and scheduler integration.

Please let me know if anyone is already working on it.

**Comment by observingClouds** [comment]
@SKM2227229725 and everyone else interested in this, please join the discussion at #107 where we already have a very good PR that just needs to account for one more training scheduler.

**Comment by SKM2227229725** [comment]
@observingClouds  Thanks for the clarification. I’ll join the discussion in #107 and look into adding the remaining training scheduler.
